controllers/appgroup_controller.go

```diff
@@ -296,6 +295,8 @@ func (r *ApplicationGroupReconciler) handleResponseAndEvent(ctx context.Context,
 		errStr = err.Error()
 		grp.ReadyFailed(errStr)
 	} else {
+		// Only update the observed generation when the reconciliation succeeds
+		grp.Status.ObservedGeneration = grp.Generation
```

Comment:
This commit seems to be causing issues - things were working fine before this. Can you verify?

Reply:
NVM pilot error - bad environment. DOH!